### ISSUE TITLE
[FEATURE REQUEST] Adding LSTM model

### ISSUE THREAD

**Issue by bagustris** [felixbur/nkululeko#256]
**Why do you propose this feature?**
LSTM is one of the most used models for sequential and time-based signals like speech. It is worth to add this model to Nkululeko

**Describe the solution you'd like**
```ini
[MODEL]
type = lstm
layers = [256, 128]
```

The easiest solution would be to adapt the implementation in https://github.com/arkiven4/NXSCough to this repo.

@arkiven4 will tackle this issue.

**Comment by felixbur** [comment]
Super, thanks!

**Comment by bagustris** [comment]
Review from copilot regarding this issue: 

What makes it a good fit

 - The codebase already has PyTorch-based ANN models (MLP, CNN) with a clean base class in model.py. An LSTM would follow the exact same pattern — inherit Model, call super().set_model_type("ann"), use the existing training/epoch loop in modelrunner.py.
 - The layers = [256, 128] config style proposed in the issue is already standard in MLP/CNN.
 - Only two files need touching: add model_lstm.py + one elif in modelrunner.py (or register via plugin registry once #33 is done).

The key challenge to be aware of

The current data pipeline outputs one aggregated feature vector per utterance (flat DataFrame row). LSTM expects sequential input (frames over time). There are two ways to handle this:

| Approach  | Feasibility  | Quality |
| ---------------- | --------------  | --------- |
| Treat each feature dim as a timestep (reshape flat vector → sequence)    | Easy, fits current   pipeline  | Limited — loses true temporal structure  |
| Frame-level features (e.g. from melspec, stft) reshaped to (batch, time, | Needs a data loader   | Proper LSTM usage   |


The MLP already partially does approach 1 (flat vector in). For proper speech LSTM, approach 2 is better — frame-level spectral features (mel spectrogram already exists as a feature extractor) reshaped into sequences.

Recommendation

Yes, add it — but implement it properly with frame-level input using existing melspec/stft features. The effort is modest since:

 1. The PyTorch training loop infrastructure is already there
 2. melspec feature extractor already exists and produces the right raw material
 3. It fills a real gap — MLP/CNN don't model temporal dynamics
